Sources/AWSLambdaRuntimeCore/Lambda.swift

```diff
@@ -118,11 +121,19 @@ public enum Lambda {
                     #if DEBUG
                     signalSource.cancel()
                     #endif
-                    eventLoop.shutdownGracefully { error in
+                    
+                    serviceLifecycle.shutdown { (error) in
                         if let error = error {
-                            preconditionFailure("Failed to shutdown eventloop: \(error)")
+                            preconditionFailure("Failed to shutdown service: \(error)")
+                        }
+                        
+                        eventLoop.shutdownGracefully { error in
```

Comment:
shouldn't that also be triggered by the lifecycle shutdown as a lifecycle task?

Reply:
I think thats a good idea: register the event loop in L110 then just call  serviceLifecycle.shutdown